code/modules/mob/living/simple_animal/hostile/cockroach.dm

```diff
@@ -0,0 +1,194 @@
+/mob/living/simple_animal/hostile/cockroach
+	name = "cockroach"
+	desc = "An exceptionally hungry pest. Keep away from food."
+	icon_state = "cockroach_new"
+	icon_living = "cockroach_new"
+	verb_say = "chitters"
+	verb_ask = "chitters inquisitively"
+	verb_exclaim = "chitters loudly"
+	verb_yell = "chitters loudly"
+	friendly_verb_simple = "nibble"
+	friendly_verb_continuous = "nibbles"
+	response_disarm_continuous = "shoos"
+	response_disarm_simple = "shoo"
+	response_harm_continuous = "splats"
+	response_harm_simple = "splat"
+	speak_emote = list("chitters")
+	deathsound = 'sound/creatures/roach_die.ogg'
+	maxHealth = 1
+	health = 1
+	//combat_mode = FALSE
+	loot = list(/obj/effect/decal/cleanable/insectguts)
+	environment_smash = ENVIRONMENT_SMASH_NONE
+	pass_flags = PASSTABLE | PASSGRILLE | PASSMOB
+	gold_core_spawnable = FRIENDLY_SPAWN
+	density = FALSE
+	mob_size = MOB_SIZE_TINY
+	mob_biotypes = MOB_ORGANIC|MOB_BUG
+	search_objects = 1
+	wanted_objects = list(/obj/item/food)
+	del_on_death = 1
+	minimum_distance = 0
+	vision_range = 4
+	robust_searching = 1 //for checking friend list
+	lose_patience_timeout = 5 SECONDS
+
+	var/burner = FALSE //is the roach on fire
+	var/sterile = FALSE //no self-reproduction
+	var/obj/item/grenade/explosive = null
+	var/explode_chance = 0
+
+/mob/living/simple_animal/hostile/cockroach/Initialize(mapload)
+	. = ..()
+	ADD_TRAIT(src, TRAIT_VENTCRAWLER_ALWAYS, INNATE_TRAIT)
+	AddComponent(/datum/component/clickbox, x_offset = -2, y_offset = -2)
+	AddComponent(/datum/component/swarming)
+	AddComponent(/datum/component/squashable, squash_chance = 35, squash_damage = 1)
+	AddComponent(/datum/component/tameable, food_types = list(/obj/item/food), tame_chance = 30, copy_faction = FALSE, bonus_tame_chance = 20)
+
+/mob/living/simple_animal/hostile/cockroach/GiveTarget(new_target)
+	add_target(new_target)
+	LosePatience()
+	if(target != null)
+		if(istype(target, /obj/item/food))
+			GainPatience()
+			Aggro()
+			retreat_distance = 0
+			minimum_distance = 0
+			return TRUE
+		else if(isliving(target))
+			GainPatience()
+			Aggro()
+			if(explosive)
+				retreat_distance = 0
+				minimum_distance = 1
+				return TRUE
+			visible_message(span_danger("[name] scuttles away!"))
+			retreat_distance = 10
+			minimum_distance = 10
+			return TRUE
+
+/mob/living/simple_animal/hostile/cockroach/AttackingTarget()
+	if(istype(target, /obj/item))
+		consume(target)
+		return
+	return ..()
+
+/mob/living/simple_animal/hostile/cockroach/proc/consume(atom/movable/fooditem)
+	playsound(src, 'sound/items/eatfood.ogg', 100, FALSE)
+	visible_message(span_danger("[name] devours [fooditem]!"))
+	qdel(fooditem)
+	if(sterile)
+		return
+	var/cap = CONFIG_GET(number/roachcap)
+	if(LAZYLEN(SSmobs.roaches) >= cap)
+		return
+	new /mob/living/simple_animal/hostile/cockroach(get_turf(src))
+
+/mob/living/simple_animal/hostile/cockroach/attackby(obj/item/I, mob/living/user)
```

Comment:
i think attackby uses `. = ..()` at the top. Consult other attackby() and tell me if im wrong pls

Reply:
It doesn't seem like others do, no. This needs to return early for certain item attacks so it can prevent the actual attack from happening.